cancermuts/datasources.py

```diff
@@ -581,6 +581,918 @@ def _get_available_mutations(self, gene_id, metadata=[]):
+                 for gm in data["genomic_annotations_obj"].values()
+                ])
+            if "clinvar_variant_id" in metadata:
+                out_metadata["clinvar_variant_id"].append(clinvar_id)
+
+            mutations.append(mutation)
+
+        # Create auxiliary dataframes:
+        df_strange = pd.DataFrame()
+        df_not_found = pd.DataFrame()
+        df2 = pd.DataFrame()
+
+        if missense_variants:
+            inconsistency_output_list = []
+
+            # Inconsistency annotations:
+            if inconsistent_annotations:
+                columns = ['variant_id',
+                            'variant_name',
+                            'genomic_annotation',
+                            'gene_name', 
+                            'interpretation', 
+                            'condition',
+                            'review_status',
+                            'methods']
+                    
+                inconsistency_output_list = self._melting_dictionary(inconsistent_annotations, add_method=True)
+                df2 = pd.DataFrame(inconsistency_output_list, columns=columns)
+
+        # Variants with uncanonical or problematic annotations:
+        if strange_variant_annotation:
+            data_strange = {}
+            for clinvar_id, information in strange_variant_annotation.items():
+                data_strange[clinvar_id] = [information["variant"], information["gene"]]
+            df_strange = pd.DataFrame.from_dict(data_strange, 
+                                                orient="index",
+                                                columns=['variant_name','gene_name'])
+            df_strange.index.name = 'clinvar_id'
+
+        # Entry not found:
+        if len(not_found_gene) != 0:
+            if len(not_found_var) != 0:
+                df_not_found = pd.DataFrame(list(zip(not_found_gene, not_found_var)), columns=['gene_name', "variant_name"])
+            else:
+                df_not_found = pd.DataFrame(list(not_found_gene), columns=['gene_name'])
+
+        return mutations, out_metadata, df_strange, df_not_found, df2, not_found_gene
+
+    def add_mutations(self, sequence, metadata=[]):
+        gene = sequence.gene_id
+        refseq = sequence.aliases["refseq"]
+        filter_missense_variants = f'({gene}[gene] AND ("missense variant"[molecular consequence] OR "SO:0001583"[molecular consequence]))'
+        mutation_type = "missense"
+
+        try:
+            clinvar_ids, out_gene = self._filtered_variants_extractor(filter_missense_variants,mutation_type,gene)
+        except RuntimeError as e:
+            raise RuntimeError(f"ClinVar mutation extraction failed: {e}")
+
+        mutations, out_metadata, df_strange, df_not_found, df2, not_found_gene = self._get_available_muts_and_md(sequence, gene, refseq, clinvar_ids, metadata)
+
+        if not clinvar_ids:
+            not_found_gene.append(out_gene)
+            return
```

Comment:
I'm a bit confused by this, shouldn't this be before the call to `self._get_available_muts_and_md`? And if so there should be at least a warning/error, if we're ending the function like that without adding anything to the Sequence object